solr/core/src/java/org/apache/solr/cloud/rule/Rule.java

```diff
@@ -365,7 +365,7 @@ public boolean equals(Object obj) {
 
     @Override
     public int hashCode() {
-      throw new UnsupportedOperationException("TODO unimplemented");
+      return Objects.hash(name, fuzzy);
```

Comment:
Given that `fuzzy` is not included in the `equals()` implementation just above, we could have two instances that are `equals()` but that have different values for `hashCode()`. I believe this is not correct.

Reply:
Just using name here.